core/shared/src/test/scala/fs2/StreamSuite.scala

```diff
@@ -550,6 +552,40 @@ class StreamSuite extends Fs2Suite {
         )
     }
 
+    test(
+      "Resource acquired on another thread pool is released when compiled stream is interrupted"
+    ) {
+      val otherECRes =
+        Resource.make(IO {
+          scala.concurrent.ExecutionContext.fromExecutorService(Executors.newCachedThreadPool())
+        })(ec =>
+          IO {
+            ec.shutdown
+          }
+        )
+      import scala.concurrent.duration._
+      def res(ref: Ref[IO, List[String]], ec: ExecutionContext): Resource[IO, Unit] =
+        Resource.make(munitContextShift.evalOn(ec)(IO {
+          println("starting acq")
+          Thread.sleep(1000)
+          println("acq")
+        }))(_ => IO(println("released")))
+
+      otherECRes.use { otherEC =>
+        (for {
+          st <- Ref.of[IO, List[String]](List.empty)
+          res <- IO.race(
+            Stream.resource(res(st, otherEC)).compile.toVector,
+            IO.sleep(100.millis)
+          )
+          result <- st.get
+          _ <- IO.sleep(1.seconds)
```

Comment:
Does this still fail if this sleep is 2 seconds?

Reply:
Yes still fails with 5 sec